sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/interceptor/HttpAsyncClient4xInterceptor.java

```diff
@@ -100,4 +127,60 @@ public ExecuteContext onThrow(ExecuteContext context) {
         LogUtils.printHttpRequestOnThrowPoint(context);
         return context;
     }
+
+    private Map<String, String> getHeaders(HttpRequest httpRequest) {
+        Map<String, String> headerMap = new HashMap<>();
+        for (Header header : httpRequest.getAllHeaders()) {
+            headerMap.putIfAbsent(header.getName(), header.getValue());
+        }
+        return headerMap;
+    }
+
+    private void handleXdsRouterAndUpdateHttpRequest(HttpRequest httpRequest, ExecuteContext context) {
+        if (!routerConfig.isEnabledXdsRoute()) {
+            return;
+        }
+        URI uri = URI.create(httpRequest.getRequestLine().getUri());
+        String host = uri.getHost();
+        if (!BaseHttpRouterUtils.isXdsRouteRequired(host)) {
+            return;
+        }
+
+        // use xds route to find a service instance, and modify url by it
+        Optional<ServiceInstance> serviceInstanceOptional = BaseHttpRouterUtils
+                .chooseServiceInstanceByXds(host.split("\\.")[0], uri.getPath(), getHeaders(httpRequest));
```

Comment:
\\. It is recommended that the value be extracted as a constant.

Reply:
done